datahub-web-react/src/app/embed/lookup/EmbedLookup.tsx

```diff
@@ -19,17 +14,29 @@ const PageContainer = styled.div`
     height: 85vh;
 `;
 
+const LookupLoading = styled(LoadingOutlined)`
+    font-size: 50px;
+`;
+
+type RouteParams = {
+    url: string;
+};
+
 const EmbedLookup = () => {
+    const history = useHistory();
     const { url: encodedUrl } = useParams<RouteParams>();
     const decodedUrl = decodeURIComponent(encodedUrl);
-    const { count, entity, error, loading } = useGetEntityByUrl(decodedUrl);
+    const { embedUrl, notFound, foundMultiple, error } = useGetEntityByUrl(decodedUrl);
+
+    useEffect(() => {
+        if (embedUrl) history.push(embedUrl);
+    }, [embedUrl, history]);
 
     const getContent = () => {
-        if (loading) return <LookupLoading />;
         if (error) return <ErrorSection />;
-        if (count === 0 || !entity) return <LookupNotFound url={encodedUrl} />;
```

Comment:
This was my logic problem, entity would be null unless there was exactly 1 search result.